app/controllers/application_controller.rb

```diff
@@ -1,3 +1,15 @@
 class ApplicationController < ActionController::Base
+  
   protect_from_forgery
-end
+
+  private
+  def current_cart
+    begin
+      Cart.find(session[:cart_id])
+    rescue ActiveRecord::RecordNotFound
+      cart = Cart.create
+      session[:cart_id] = cart.id
+      cart
+    end
```

Comment:
use this 
cart = Cart.find_by_id(session[:cart_id])
 unless cart 
        cart = Cart.create 
        session[:cart_id] = cart.id
end
cart